app/controllers/api/deploy_groups_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+class Api::DeployGroupsController < Api::BaseController
+  include DeployGroupsCommon
+  skip_before_action :require_project, only: :index
+  before_action :enabled?
+
+  def index
+    render json: paginate(deploy_groups)
+  end
+
+  def project_deploy_groups
+    render json: paginate(stage.deploy_groups)
+  end
+
+  def update_deploy_groups
+    if !production_change?
+      stage.deploy_groups = deploy_groups_to_add
+      stage.save!
+      head :no_content
+    else
+      render json: {message: "Do not modify production via the API"}, status: 403
```

Comment:
hmm. I guess we could get rid of it. It was an attempt to protect stages from modification via the API if they are production.